ConsoleApp2/ConsoleApp2/DelegAndEventsExamples.cs

```diff
@@ -0,0 +1,128 @@
+﻿using System;
+
+namespace ConsoleApp2
+{
+    class DelegAndEventsExamples
+    {
+        delegate void Message();
+
+        public void Example()
+        {
+            Message mes;
+            if (DateTime.Now.Hour < 12)
+            {
+                mes = GoodMorning;
+            }
+            else
+            {
+                mes = GoodEvening;
+            }
+            mes();
+            mes += GoodEvening;
+            mes();
+            //Action
+            Action<int, int> op;
+
+            op = Add;
+            Operation(10, 6, op);
+            op = Substract;
+            Operation(10, 6, op);
+
+            //Predicate
+            Predicate<int> isPositive = delegate (int x) { return x > 0; };
+            Console.WriteLine(isPositive(20));
+            Console.WriteLine(isPositive(-20));
+
+            //Func
+            Func<int, int> retFunc = Factorial;
+            int n1 = GetInt(6, retFunc);
+            Console.WriteLine(n1);  // 720
+            int n2 = GetInt(6, x => x * x);
+            Console.WriteLine(n2); // 36
+
+            Account acc = new Account(100);
+            acc.Notify += DisplayMessage;   // Добавляем обработчик для события Notify
+            acc.Put(20);    // добавляем на счет 20
+            Console.WriteLine($"Сумма на счете: {acc.Sum}");
+            acc.Take(70);   // пытаемся снять со счета 70
+            Console.WriteLine($"Сумма на счете: {acc.Sum}");
+            acc.Take(180);  // пытаемся снять со счета 180
+            Console.WriteLine($"Сумма на счете: {acc.Sum}");
+            Console.Read();
+
+        }
+
+        static void Operation(int x1, int x2, Action<int, int> op)
+        {
+            if (x1 > x2)
+                op(x1, x2);
+        }
+        static void Add(int x1, int x2)
+        {
+            Console.WriteLine("Сумма чисел: " + (x1 + x2));
```

Comment:
method actually adds and displays result
better to split those operations

Reply:
done